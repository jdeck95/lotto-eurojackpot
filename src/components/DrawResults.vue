<script setup lang="ts">
interface Props {
  numbers: string[];
  additionalNumbers: string[];
}

const props = defineProps<Props>();
</script>

<template>
  <div
    v-for="number in props.numbers"
    :key="number"
    class="draw-number"
  >
    {{ number }}
  </div>
  <div class="column justify-center q-mx-md">+</div>
  <div
    v-for="additionalNumber in props.additionalNumbers"
    :key="additionalNumber"
    class="draw-number draw-number--additional"
  >
    {{ additionalNumber }}
  </div>
</template>

<style scoped lang="scss">
.draw-number {
  border: 1px solid grey;
  border-radius: 50%;
  text-align: center;
  line-height: 16px;
  padding: 16px;
  width: 50px;
  font-weight: 800;

  &--additional {
    background-color: #ffd535;
    border: 0px;
  }
}
</style>
