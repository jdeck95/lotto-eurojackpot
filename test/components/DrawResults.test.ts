import { mount, type VueWrapper } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { setActivePinia } from 'pinia';
import { Quasar } from 'quasar';
import { beforeEach, describe, expect, it } from 'vitest';
import DrawResults from '../../src/components/DrawResults.vue';

describe('DrawResults', () => {
  let wrapper: VueWrapper<any>;

  const pinia = createTestingPinia();
  setActivePinia(pinia);

  beforeEach(() => {
    wrapper = mount(DrawResults, {
      global: {
        plugins: [Quasar, pinia],
      },
      props: {
        numbers: ['1', '2', '3', '4', '5', '6'],
        additionalNumbers: ['1', '2'],
      },
    });
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
