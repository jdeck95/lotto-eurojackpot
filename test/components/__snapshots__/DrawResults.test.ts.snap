// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DrawResults > should render correctly 1`] = `
"<div data-v-c506f757="" class="draw-number q-mx-xs">1</div>
<div data-v-c506f757="" class="draw-number q-mx-xs">2</div>
<div data-v-c506f757="" class="draw-number q-mx-xs">3</div>
<div data-v-c506f757="" class="draw-number q-mx-xs">4</div>
<div data-v-c506f757="" class="draw-number q-mx-xs">5</div>
<div data-v-c506f757="" class="draw-number q-mx-xs">6</div>
<div data-v-c506f757="" class="column justify-center q-mx-md">+</div>
<div data-v-c506f757="" class="draw-number draw-number--additional q-mx-xs">1</div>
<div data-v-c506f757="" class="draw-number draw-number--additional q-mx-xs">2</div>"
`;
